File: java/vauvau.java
import java.util.Scanner;

public class vauvau {
	static void getAttack(int[] attacks, int attack, int rest) {
		boolean isAttack = true;
		int count = attack;
		for (int i = 0; i < attacks.length; i++) {
			if (isAttack)
				attacks[i] += 1;
			if (--count == 0) {
				isAttack = !isAttack;
				count = isAttack ? attack : rest;
			}
		}
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int A = scan.nextInt();
		int B = scan.nextInt();
		int C = scan.nextInt();
		int D = scan.nextInt();
		int P = scan.nextInt();
		int G = scan.nextInt();
		int M = scan.nextInt();
		int maxTime = Math.max(P, Math.max(G, M));
		int[] attacks = new int[maxTime];
		getAttack(attacks, A, B);
		getAttack(attacks, C, D);
		for (int i : new int[] { P, G, M }) {
			switch (attacks[i - 1]) {
			case 0:
				System.out.println("none");
				break;
			case 1:
				System.out.println("one");
				break;
			default:
				System.out.println("both");
				break;
			}
		}
		scan.close();
	}
}

File: ts/src/vauvau.ts
import readline = require('readline')

interface Dog {
  barkTime: number
  restTime: number
}

function getAttackList (dogs: Dog[], maxTime: number): number[] {
  let result = Array<number>(maxTime).fill(0)
  dogs.forEach(dog => {
    result = getDogAttacks(result, dog)
  })
  return result
}

function getDogAttacks (list: number[], dog: Dog): number[] {
  const result: number[] = []

  let isAttack = true
  let count = dog.barkTime
  for (let i = 0; i < list.length; i++) {
    result[i] = (list[i] ?? 0) + (isAttack ? 1 : 0)
    if (--count === 0) {
      isAttack = !isAttack
      count = isAttack ? dog.barkTime : dog.restTime
    }
  }

  return result
}

const dogs: Dog[] = []
let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount === 0) {
      const [A, B, C, D] = line.split(' ').map(Number)
      dogs.push({ barkTime: A ?? 0, restTime: B ?? 0 })
      dogs.push({ barkTime: C ?? 0, restTime: D ?? 0 })
    } else {
      const arrival = line.split(' ').map(Number)
      const attacks = getAttackList(dogs, Math.max(...arrival))
      arrival.forEach((time) => {
        switch (attacks[time - 1]) {
          case 0:
            console.log('none')
            break
          case 1:
            console.log('one')
            break
          default:
            console.log('both')
            break
        }
      })
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
